src/components/SelectNumber/index.tsx

```diff
@@ -0,0 +1,354 @@
+
+const roundStep = (value: number, step: number, direction: number) =>
+  direction === -1
+    ? Math.floor(value / step) * step
+    : Math.ceil(value / step) * step
+
+const disabledStyles = ({
+  disabled,
+  theme,
+}: {
+  disabled: boolean
+  theme: Theme
+}) =>
+  disabled &&
+  `
+    background-color: ${theme.colors.neutral.backgroundWeakDisabled};
+    border: none;
+    color: ${theme.colors.neutral.textWeakDisabled};
+    opacity: 1;
+    cursor: not-allowed;
+  `
+
+const containerSizes = {
+  large: 48,
+  medium: 40,
+  small: 32,
+}
+
+const centerBoxSizes = {
+  large: 32,
+  medium: 32,
+  small: 24,
+}
+
+type ContainerSizesType = keyof typeof containerSizes
+export const containerSizesKeys = Object.keys(
+  containerSizes,
+) as ContainerSizesType[]
+
+const iconSizes = {
+  large: 26,
+  medium: 24,
+  small: 22,
+}
+
+const StyledTouchable = styled(Touchable, {
+  shouldForwardProp: prop => !['position', 'size'].includes(prop.toString()),
+})<{ size: ContainerSizesType }>`
+  justify-content: center;
+  align-items: center;
+  position: relative;
+  border-radius: ${({ theme }) => theme.radii.default};
+  width: ${({ size }) => centerBoxSizes[size]}px;
+  height: ${({ size }) => centerBoxSizes[size]}px;
+  color: ${({ theme, disabled }) =>
+    disabled
+      ? theme.colors.neutral.textWeakDisabled
+      : theme.colors.primary.textWeak};
+
+  :hover:not([disabled]) {
+    background: ${({ theme }) => theme.colors.primary.background};
+  }
+
+  margin: 0 ${({ theme }) => theme.space['1']};
+`
+
+const StyledCenterBox = styled('div', {
+  shouldForwardProp: prop => !['size'].includes(prop.toString()),
+})<{ size: ContainerSizesType }>`
+  display: flex;
+  flex: 1;
+  flex-direction: row;
+  height: ${({ size }) => centerBoxSizes[size]}px;
+  align-items: center;
+  outline: none;
+  justify-content: center;
+  border-radius: ${({ theme }) => theme.radii.default};
+  border: 1px solid transparent;
+  :hover:not([disabled], :focus) {
+    border: 1px solid ${({ theme }) => theme.colors.primary.borderWeakHover};
+  }
+  :focus-within:not([disabled]) {
+    box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
+    border: 1px solid ${({ theme }) => theme.colors.primary.borderWeakHover};
+  }
+  max-width: calc(100% - ${({ size }) => containerSizes[size] * 2}px);
+`
+
+const StyledInput = styled.input`
+  color: ${({ theme }) => theme.colors.neutral.text};
+  background-color: transparent;
+  font-size: 16;
+  border: none;
+  text-align: right;
+  outline: none;
+  position: relative;
+  margin-right: 4;
+  max-width: 100%;
+  font-weight: 500;
+  text-align: center;
+`
+
+const StyledText = styled('span', {
+  shouldForwardProp: prop => !['disabled'].includes(prop.toString()),
+})<{ disabled: boolean }>`
+  color: ${({ theme, disabled }) =>
+    disabled
+      ? theme.colors.neutral.textWeakDisabled
+      : theme.colors.neutral.text};
+  user-select: none;
+`
+
+const StyledContainer = styled(Box, {
```

Comment:
Why ? If it's a brand new component we should start clean no ?

Reply:
Yes please remove box and when merging on shire you can remove all box props and use styled